feat: template crd (#4302)
kubeshop/testkube: docs/docs/articles/templates.mdx

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

# Templates

> Templates allow you to store templates for other resources used in Testkube. We support such a list of templates job | container | cronjob | scraper | pvc | webhook. To define templates in Testkube, you'll need to provide a template body (in golang template format) and a type of the template.

Reviewer: ```suggestion
Templates allow you to store templates for other resources used in Testkube. We support a list of templates job | container | cronjob | scraper | pvc | webhook. To define templates in Testkube, you'll need to provide a template body (in Golang template format) and a type of the template.
```

## Creating a Template
The template can be created using the API, CLI, or a Custom Resource.

<Tabs>
<TabItem value="api" label="API">

> If you prefer to use the API for creating a template, you can check API spec for templates in below doc.

Reviewer: ```suggestion
If you prefer to use the API for creating a template, please visit the API spec for templates in the doc below.
```

![OpenAPI spec](../openapi.md)

</TabItem>

<TabItem value="cli" label="CLI">

> Templates can be created with Testkube CLI using the `create template` command.

Reviewer: ```suggestion
Templates can be created with the Testkube CLI using the `create template` command.
```

```sh
kubectl testkube create template --name job-template --template-type job --body job-template.yaml
```

`--name` - Your template name (in this case `job-template`).
`--template-type` - Your template type (in this case `job` for prebuilt executors).
`--body` - A path to the file with job template content

</TabItem>

<TabItem value="crd" label="Custom Resource">

```yaml title="template.yaml"
apiVersion: tests.testkube.io/v1
kind: Webhook
metadata:
  name: example-webhook
  namespace: testkube
spec:
  type: job
  body: <YOUR_TEMPLATE_BODY>
```

> Where <YOUR_TEMPLATE_BODY> should be replaced with kubernetes job definition in golang template format.

Reviewer: ```suggestion
Where <YOUR_TEMPLATE_BODY> should be replaced with the Kubernetes job definition in Golang template format.
```

And then apply with:

```sh
kubectl apply -f template.yaml
```

</TabItem>

</Tabs>

> ### Using template

Reviewer: ```suggestion
### Using Templates
```

> You will need to refer a template in corresponding reference field of the resource.

Reviewer: ```suggestion
You will need to refer to a template in the corresponding reference field of the resource.
```

<Tabs>
<TabItem value="api" label="API">

> Check templateReference fields in api spec, for example, Test -> executionRequest -> jobTemplateReference field

Reviewer: ```suggestion
Check templateReference fields in API spec. For example, Test -> executionRequest -> jobTemplateReference field.
```

![OpenAPI spec](../openapi.md)

</TabItem>

<TabItem value="cli" label="CLI">

> Templates can be created with Testkube CLI using the `create template` command.

Reviewer: ```suggestion
Templates can be created with the Testkube CLI using the `create template` command.
```
